src/inspector_agent.cc

```diff
@@ -648,6 +659,101 @@ void Agent::AllAsyncTasksCanceled() {
   client_->AllAsyncTasksCanceled();
 }
 
+void Agent::InterceptAsyncStackDepthMessage(const StringView& message) {
+  // This logic would be better implemented in JavaScript, but when using
+  // --inspect-brk, the debugger must necessarily attach before much JavaScript
+  // can execute. The Debugger.setAsyncCallStackDepth message arrives too early
+  // and we must intercept this in C++.
+
+  v8::Isolate* isolate = parent_env_->isolate();
+  Local<Context> context = parent_env_->context();
+
+  MaybeLocal<String> string =
+    String::NewFromTwoByte(isolate, message.characters16(),
+                           v8::NewStringType::kNormal, message.length());
+  if (string.IsEmpty()) {
```

Comment:
I can’t really tell how relevant performance is here, but if it makes sense to you, maybe add an early return if `message` doesn’t contain the substring `Debugger.setAsync`?

Reply:
I think it would be good to improve performance here, but that's a non-critical optimization and can be addressed once this lands. I have left a TODO.